python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -20,179 +20,142 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 import numpy as np
-from scarlet.source import PointSource, ExtendedSource, SourceInitError
-from scarlet.component import BlendFlag, Component
+from scarlet.source import PointSource, ExtendedSource, MultiComponentSource
 
 import lsst.afw.image as afwImage
 from lsst.afw.geom import SpanSet
 from lsst.geom import Point2I
+import lsst.log
 import lsst.afw.detection as afwDet
 
-__all__ = ["LsstSource", "LsstHistory"]
-
-
-class BlankSource(Component):
-    """Placeholder for a source that cannot be initialized
-
-    In DM-19790 this class will be removed when skipped sources are
-    not deblended at all.
+__all__ = ["init_source", "morphToHeavy", "modelToHeavy"]
+
+logger = lsst.log.Log.getLogger("meas.deblender.deblend")
+
+
+def init_source(frame, peak, observation, bbox,
+                symmetric=False, monotonic=True,
+                thresh=5, components=1):
+    """Initialize a Source
+
+    The user can specify the number of desired components
+    for the modeled source. If scarlet cannot initialize a
+    model with the desired number of components it continues
+    to attempt initialization of one fewer component until
+    it finds a model that can be initialized. If all of the
+    models fail, including a `PointSource` model, then skip
+    the source.
+
+    Parameters
+    ----------
+    frame: `LsstFrame`
+        The model frame for the scene
+    peak: `PeakRecord`
+        Record for a peak in the parent `PeakCatalog`
+    observation: `LsstObservation`
+        The images, psfs, etc, of the observed data.
+    bbox: `Rect`
```

Comment:
I don't think you imported Rect anywhere so this probably won't work.

Reply:
It should, because I don't think this actaully creates a link to the class. Otherwise I'd have to import `Rect` but never use it, which would cause `flake8` to fail.

Reply:
I thought the whole point was to create links, but apparently it doesn't? I'll have to ask about that.

Anyway, in this case since you're not actually importing Rect, the full namespace would be more useful.

Reply:
Ok. It's a good thing you had me do this, I forgot that the stack doesn't call it `Rect`, it's `Box2I`.